canonicalwebteam/blog/wordpress_api.py

```diff
@@ -19,71 +21,74 @@ def process_response(response):
     return response.json()
 
 
-def build_get_articles_url(
-    tags=[],
-    per_page=12,
-    page=1,
-    tags_exclude=[],
-    exclude=[],
-    categories=[],
-    sticky="",
-    groups=[],
-    after="",
-    before="",
-    author="",
-):
+def build_url(endpoint, params):
     """
     Build url to fetch articles from Wordpress api
-    :param tags: Array of tag ids to fetch articles for
-    :param per_page: Articles to get per page
-    :param page: Page number to get
-    :param tags_exclude: Array of IDs of tags that will be excluded
-    :param exclude: Array of article IDs to be excluded
-    :param category: Array of categories, which articles
-        should be fetched for
-    :param sticky: string 'true' or 'false' to only get featured articles
-    :param before: ISO8601 compliant date string to limit by date
-    :param after: ISO8601 compliant date string to limit by date
+    :param endpoint: The REST endpoint to fetch data from
+    :param params: Dictionary of parameter keys and their values
 
     :returns: URL to Wordpress api
     """
-    url = (
-        f"{API_URL}/posts?per_page={per_page}"
-        f"&tags={','.join(str(id) for id in tags)}"
-        f"&page={page}"
-        f"&group={','.join(str(id) for id in groups)}"
-        f"&tags_exclude={','.join(str(id) for id in tags_exclude)}"
-        f"&categories={','.join(str(id) for id in categories)}"
-        f"&exclude={','.join(str(id) for id in exclude)}"
-        f"&author={author}&_embed"
-    )
-    if sticky != "":
-        url = url + f"&sticky={sticky}"
-    if before != "":
-        url = url + f"&before={before}"
-    if after != "":
-        url = url + f"&after={after}"
 
-    return url
+    clean_params = [(key, value) for key, value in params.items() if value]
```

Comment:
You do this to filter out items whose value is `None` or `False`? Is this actually a case you're dealing with, and are you sure that's what you want to do? This seems to add a lot of complexity.

I would have thought you'd expect people using this function to know what they're doing and be passing through each parameter deliberately, they wouldn't expect some parameters to be filtered out. So either, do something special with `None` or `False`, or just do nothing and let them be converted into strings naturally. What you're doing here feels like over-engineering for a case you probably don't have.

Even if you did want to do this, it's a shame to convert a relatively easy to use `dict` into a confusing list of tuples. You should keep it as a dictionary which would be much easier to work with in your following for loop. You could do this through a dictionary comprehension:

``` python3
clean_params = {key: value for key, value in params.items() if value}
```

But as I say, in this case I'd probably suggest not doing it at all.

Reply:
Hm yeah, I was on this for a bit. The problem with explicitely matching `None` is that I wont be able to exclude all the empty strings.

Ill use lamda funtions here. They dont  flow so well in Python but should solve this nicely.

Reply:
Updated it. 

I feel like funtional programming is just not well supported, but the flow seems much cleaner here.

Would love to chain the function without nesting in parantheses. Is there a way to do it in Python?